simtools/utils/names.py

```diff
@@ -823,3 +824,35 @@ def translate_simtools_to_corsika(simtools_par):
             {simtools_par}."
         _logger.debug(msg)
         return simtools_par
+
+
+def sanitize_name(name):
+    """
+    Sanitize name to be a valid Python identifier.
+
+    - Replaces spaces with underscores
+    - Converts to lowercase
+    - Removes characters that are not alphanumerics or underscores
+    - If the name starts with a number, prepend an underscore
+
+    Parameters
+    ----------
+    name: str
+        name to be sanitized.
+
+    Returns
+    -------
+    str:
+        Sanitized name.
+    """
+
+    # Convert to lowercase
+    sanitized = name.lower()
+
+    # Replace spaces with underscores
+    sanitized = sanitized.replace(" ", "_")
+
+    # Remove characters that are not alphanumerics or underscores
+    sanitized = re.sub(r"\W|^(?=\d)", "_", sanitized)
+
```

Comment:
At the moment I think it is not necessary. I added this because I was getting lots of warnings from the Tables module that I wouldn't be able to use dot ('.') to access attributes if the attribute names are not sanitized. That means, if the attribute name is not sanitized, we get a warning. And the name of the attributes are fixed (should not be changed).

Reply:
But are you sure that you really catching all cases of input strings to sanitize them? Adding a "if not sanitized.isidentifier() raise WhateverError" would make this function more robust.

Reply:
I added now as you suggested (I didnt know about the "isidentifier" method). however, I do not know how to include a test, since I do expect that every string would be covered by the function.

Reply:
OK!